gems/smithy-client/spec/support/client_helper.rb

```diff
@@ -43,54 +63,55 @@ def sample_shapes
             'byte' => { 'target' => 'smithy.api#Byte' },
-            # 'document' => { 'target' => 'smithy.api#Document' },
+            'document' => { 'target' => 'smithy.api#Document' },
             'double' => { 'target' => 'smithy.api#Double' },
             'enum' => { 'target' => 'smithy.ruby.tests#Enum' },
             'float' => { 'target' => 'smithy.api#Float' },
-            'integer' => { 'target' => 'smithy.api#Integer' },
             'intEnum' => { 'target' => 'smithy.ruby.tests#intEnum' },
+            'integer' => { 'target' => 'smithy.api#Integer' },
+            'list' => { 'target' => 'smithy.ruby.tests#List' },
             'long' => { 'target' => 'smithy.api#Long' },
+            'map' => { 'target' => 'smithy.ruby.tests#Map' },
             'short' => { 'target' => 'smithy.api#Short' },
+            'streamingBlob' => {
+              'target' => 'smithy.ruby.tests#StreamingBlob',
+              'traits' => { 'smithy.api#default' => 'streamingBlob' }
+            },
             'string' => { 'target' => 'smithy.api#String' },
-            'timestamp' => { 'target' => 'smithy.api#Timestamp' },
             'structure' => { 'target' => 'smithy.ruby.tests#Structure' },
-            'list' => { 'target' => 'smithy.ruby.tests#List' },
-            'map' => { 'target' => 'smithy.ruby.tests#Map' }
-            # 'union' => { 'target' => 'smithy.api#String' }
+            'structureList' => { 'target' => 'smithy.ruby.tests#StructureList' },
+            'structureMap' => { 'target' => 'smithy.ruby.tests#StructureMap' },
+            'timestamp' => { 'target' => 'smithy.api#Timestamp' },
+            'union' => { 'target' => 'smithy.ruby.tests#Union' }
           }
         },
-        'smithy.ruby.tests#Operation' => {
-          'type' => 'operation',
-          'input' => { 'target' => 'smithy.ruby.tests#Structure' },
-          'output' => { 'target' => 'smithy.ruby.tests#Structure' }
-        },
-        'smithy.ruby.tests#SampleClient' => {
-          'type' => 'service',
-          'operations' => [
-            { 'target' => 'smithy.ruby.tests#Operation' }
-          ]
+        'smithy.ruby.tests#Union' => {
+          'type' => 'union',
+          'members' => {
+            'string' => { 'target' => 'smithy.api#String' },
+            'structure' => { 'target' => 'smithy.ruby.tests#Structure' }
+          }
         }
       }
     end
 
     def sample_client(options = {})
+      module_name = options[:module_name] || next_sample_module_name
       model = options[:model] ||= model(options)
-      namespace = options[:gem_module] || next_sample_module_name
-      plan = create_plan(model, namespace, options)
+      plan = create_plan(module_name, model, options)
       source = Smithy.source(plan)
       Object.module_eval(source)
-      Object.const_get(namespace)
-    rescue Exception => e # rubocop:disable Lint/RescueException
+      Object.const_get(module_name)
+    rescue LoadError => e
```

Comment:
Does LoadError cover all of the syntax errors that could happen during eval?

Reply:
I think so. At this point we assume generated code is valid.